components/rendering/src/markdown/codeblock.rs

```diff
@@ -22,16 +22,27 @@ pub struct CodeBlock<'config> {
 }
 
 impl<'config> CodeBlock<'config> {
-    pub fn new(fence_info: &str, config: &'config Config, background: IncludeBackground) -> Self {
+    pub fn new(fence_info: &str, config: &'config Config, background: IncludeBackground, path: Option<&'config str>) -> Self {
         let fence_info = FenceSettings::new(fence_info);
         let theme = &THEME_SET.themes[config.highlight_theme()];
-        let (highlighter, in_extra) = get_highlighter(fence_info.language, config);
+        let (highlighter, highlight_source) = get_highlighter(fence_info.language, config);
+        let extra_syntax_set = match highlight_source {
+            HighlightSource::Extra => config.markdown.extra_syntax_set.as_ref(),
+            HighlightSource::NotFound => {
+                // Language was not found, so it exists (safe unwrap)
+                let lang = fence_info.language.unwrap();
+                if let Some(path) = path {
+                    eprintln!("Warning: Highlight language {} not found in {}", lang, path);
+                } else {
+                    eprintln!("Warning: Highlight language {} not found", lang);
+                }
```

Comment:
Should we error instead? There's no point putting a lang that we don't support so it's either a missing highlighting or a typo, both of those could be considered as errors.

Reply:
There's usecase for language that is not highlighted by Zola but is still styled by CSS using the data-lang attribute of the code tag. Can be used to differentiate lyrics, poetry, quotations, etc.. So error is not a universal answer though it is the good answer for a programming blog. Adding error here would be breaking change, especially for people who use markdown shared with other programs than zola who may produce/want such information.

Maybe we can open a discussion about errors/warnings on the forums to have feedback for the next releases, not just about this specific case. In the meantime, what to do? It's not hard to add a config setting like `markdown.missing_codeblock_lang` which could be "error", "warn" or "off". But do you think it's a good idea? Otherwise, i'd be tempted to just remove the warning until we have more feedback.

Reply:
Good point, let's keep it as warning for now. We can add config to disable those warnings later if needed.